feat(Breakpoint): Add hook for getting updates on breakpoint changes

aDogCalledSpot opened the pull request:
Adds helper  functions for getting the current breakpoint and for listening to changes in breakpoints.

Example uses:

```rust
let breakpoint =  use_breakpoint();
let button = use_memo(breakpoint.clone(), |breakpoint| match breakpoint {
    Breakpoint::None => html!(<Button icon={Icon::Upload} {onclick} />,
    _ => html!(<Button icon={Icon::Upload} {onclick} label="Select from disk" variant={ButtonVariant::Secondary} />
});
```

```rust
<Page {sidebar} open={Breakpoint::current() > Breakpoint::Medium} />
```

ctron commented:
Cool! That could actually be something for `yew-more-tools`. Works here too.

ctron reviewed (changes requested):
Sorry for changing my mind.

I think if would be better if we had an explicit `get_breakpoint` function. Vs the `repr(usize)`.

aDogCalledSpot commented:
> Cool! That could actually be something for `yew-more-tools`.

I don't think so since the actual pixel width values of the breakpoints aren't universal and vary from framework to framework.

> I think if would be better if we had an explicit get_breakpoint function. Vs the `repr(usize)`.

I'm not sure I understand your suggestion. I added `Breakpoint::current()` for retrieving the current breakpoint.
Do you maybe mean adding a function like `const fn pixel_width(self) -> usize` to get the pixel values from a function instead of encoding them directly into the enum?

I added the integer values to add clarity as to how `Ord` works - the `repr(usize)` is superfluous and can be deleted. I personally don't like simply deriving `Ord` on enums that don't assign integer values since I don't feel as if the usual declaration of enums suggests an order (personal opinion). Manually implementing `Ord` for enums is a pain since you need to compare every possible pair of values and that's just hard to read. There's a very clear connection between the pixels values and the breakpoint so this feels like a good place to add it.

ctron commented:
> Do you maybe mean adding a function like const fn pixel_width(self) -> usize to get the pixel values from a function instead of encoding them directly into the enum?

Yes, that was my idea. The main reason for doing so is that one does not need to rely on "the value" (of the enum), but there's a proper function to call. Which makes this more explicit (IMHO).

> I don't think so since the actual pixel width values of the breakpoints aren't universal and vary from framework to framework.

I feared out wouldn't be that simple :D … However, give a few bonus points, it could be possible to some up with a trait helping out.

Then again, that's extra work. I know! Forget about it :wink:
